packages/react-icons/src/utils/createFluentIcon.ts

```diff
@@ -0,0 +1,31 @@
+import * as React from "react";
+import { FluentIconsProps } from "./FluentIconsProps.types";
+import { useIconState } from "./useIconState";
+
+export type FluentIcon = {
+    (props: FluentIconsProps<React.SVGAttributes<SVGElement>>): JSX.Element;
+    displayName: string | undefined;
+}
+
+export const createFluentIcon = (displayName: string, width: string, paths: string[]): FluentIcon => {
+    const viewBoxWidth = width === "1em" ? "20" : width;
+    const Icon = (props: FluentIconsProps) => {
+        props = {
+            ...useIconState(props), // HTML attributes/props for things like accessibility can be passed in, and will be expanded on the svg object at the start of the object
+            width, height: width, viewBox: `0 0 ${viewBoxWidth} ${viewBoxWidth}`, xmlns: "http://www.w3.org/2000/svg"
+        };
+        return React.createElement(
+            "svg",
+            props,
+            ...paths.map((d) =>
+                React.createElement("path", {
+                    d,
+                    fill: props.fill, // We are designating primaryFill as the primary color for filling. If not provided, it defaults to null
+                    // key: i // The key for static children is needless 
+                })
+            )
+        );
+    }
+    Icon.displayName = displayName;
+    return Icon;
+}
```

Comment:
You could maybe simplify it into this?
```suggestion
export const createFluentIcon = (
  displayName: string,
  width: string,
  paths: string[]
): FluentIcon => {
  const viewBoxWidth = width === "1em" ? "20" : width;
  const Icon = (props: FluentIconsProps) => {
    const state = useIconState({
      ...props,
      width,
      height: width,
      viewBox: `0 0 ${viewBoxWidth} ${viewBoxWidth}`,
      xmlns: "http://www.w3.org/2000/svg",
    });

    return React.createElement(
      "svg",
      state,
      ...paths.map((d) =>
        React.createElement("path", {
          d,
          fill: state.fill, // We are designating primaryFill as the primary color for filling. If not provided, it defaults to null
        })
      )
    );
  };
  Icon.displayName = displayName;
  return Icon;
};
```

Reply:
What's the key-point, you prefer `const state = xxx`? If so, this:

```js
const state = {
  ...useIconState(props),
  width,
  height: width,
  viewBox: `0 0 ${viewBoxWidth} ${viewBoxWidth}`,
  xmlns: "http://www.w3.org/2000/svg",
};
```

May be better. `useIconState` seems used to add accessibility attrs, the props like width and viewBox is only for svg element and does not need to be passed into `useIconState`.

Reply:
Mainly for readability and to not modify the props directly. You can change it to the snippet you added above, both work 🙂

Reply:
Solved.